### ISSUE TITLE
problem in print_model_code() and write_model()

### ISSUE THREAD

**Issue by rosemarychen99** [pharmpy/pharmpy#2974]
### Bug report

Pharmpy version:0.109.0

Cannot not get nonmem model code using the two functions. it looks like this. the model was created by ´mod <- create_basic_pk_model(administration="oral")´
<img width="1069" alt="image" src="https://github.com/pharmpy/pharmpy/assets/35337934/be7739a1-70f6-41bf-9118-685db29e5100">

**Comment by stellabelin** [comment]
Hi! This is because it hasn't been converted to a NONMEM model, if you use `convert_model()` it should work.

**Comment by stellabelin** [comment]
Either that or have a nicer string to represent the Pharmpy model. Was thinking since in `repr` and similar we have `<Pharmpy model object {self.name}>`, otherwise we have no way to get the overview of the object right? This would also be relevant for modelbuilder.

**Comment by rikardn** [comment]
I agree that an overview printout is wanted, but I think that the concept of `model_code` should be reserved for the external model code. Currently the json is the generic `model_code` so it makes sense to output it here, but it is at the same time a source of confusion.

**Comment by stellabelin** [comment]
True, and it also makes sense that you should be able to print the Pharmpy overview no matter which external tool is behind the model, so maybe then it makes sense to in that case have a separate function for this and raise in `print_model_code()` if it's a generic model object?
